atomic_reactor/core.py

```diff
@@ -272,10 +277,60 @@ def __init__(self, base_url=None, retry_times=DOCKER_MAX_RETRIES,
 
         if hasattr(docker, 'AutoVersionClient'):
             client_kwargs['version'] = 'auto'
-        client_kwargs['retry'] = retry_times
+        self.retry_times = retry_times
+        client_kwargs['retry'] = self.retry_times
 
         self.d = WrappedDocker(**client_kwargs)
 
+    def retry_generator(self, fnc_name, function, *args, **kwargs):
+        retry_times = int(kwargs.pop('retry_times', self.retry_times))
+        retry_delay = DOCKER_BACKOFF_FACTOR
+        retry_client_statuses = DOCKER_CLIENT_STATUS_RETRY
+
+        for counter in range(retry_times + 1):
+            exc = None
+            try:
+                logs_gen = function(*args, **kwargs)
+                cmd_result = atomic_reactor.util.wait_for_command(logs_gen)
+
+            except ProtocolError as e:
+                if counter != retry_times:
+                    exc = e
+                else:
+                    if hasattr(exc, 'message'):
```

Comment:
`Exception.message` is [deprecated](https://www.python.org/dev/peps/pep-0352/#transition-plan) and not available in py3, we shouldn't use it

Reply:
ahh ok will use .args